src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -17,17 +17,19 @@
 <kd-help-section>
   <md-input-container class="md-block" md-is-error="ctrl.isNameError()">
     <label>App name</label>
-    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required kd-unique-name
-        ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }">
-    <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate"
-        ng-show="ctrl.form.name.$pending">
-    </md-progress-linear>
-    <ng-messages for="ctrl.form.name.$error" role="alert" multiple>
-      <ng-message when="required">Application name is required.</ng-message>
-      <ng-message when="uniqueName">
-        Application with this name already exists within namespace <i>{{ctrl.namespace}}</i>.
-      </ng-message>
-    </ng-messages>
+    <input ng-model="ctrl.name"
+           name="name"
+           namespace="ctrl.namespace"
+           required
+           kd-unique-name
+           ng-pattern="/^[a-z]+[a-z\-]*[a-z]*$/">
+
+    <div ng-messages="ctrl.form.name.$error" ng-if="ctrl.form.name.$touched">
```

Comment:
This `ng-if="ctrl.form.name.$touched"` changes the behavior that is default for angular material. Normally, when you type the value in the input for the first time, you don't get errors until you're done with typing your value. This makes sense, because you don't want to get message like "this is invalid email" before you've even typed `@`. What do you think about reverting this? And staying with material design defaults. 

Note that on line 18 there is `md-is-error="ctrl.isNameError()"` that shows unique-name errors nevertheless. This is because unique-name errors are async.